app/scripts/controllers/metametrics-data-deletion/metametrics-data-deletion.ts

```diff
@@ -0,0 +1,189 @@
+import {
+  BaseController,
+  RestrictedControllerMessenger,
+} from '@metamask/base-controller';
+import type { DataDeletionService } from '../../services/data-deletion-service';
+import { DeleteRegulationStatus } from '../../../../shared/constants/metametrics';
+
+// Unique name for the controller
+const controllerName = 'MetaMetricsDataDeletionController';
+
+/**
+ * Get a type representing the public interface of the given type. The
+ * returned type will have all public properties, but will omit private
+ * properties.
+ *
+ * @template Interface - The interface to return a public representation of.
+ */
+type PublicInterface<Interface> = Pick<Interface, keyof Interface>;
+
+/**
+ * Timestamp at which regulation response is returned.
+ */
+export type DataDeleteTimestamp = number;
+/**
+ * Regulation Id retuned while creating a delete regulation.
+ */
+export type DataDeleteRegulationId = string | null;
+
+/**
+ * MetaMetricsDataDeletionController controller state
+ * metaMetricsDataDeletionId - Regulation Id retuned while creating a delete regulation.
+ * metaMetricsDataDeletionTimestamp - Timestamp at which the most recent regulation is created/requested for.
+ * metaMetricsDataDeletionStatus - Status of the current delete regulation.
+ */
+export type MetaMetricsDataDeletionState = {
+  metaMetricsDataDeletionId: DataDeleteRegulationId;
+  metaMetricsDataDeletionTimestamp: DataDeleteTimestamp;
+  metaMetricsDataDeletionStatus?: DeleteRegulationStatus;
+};
+
+const defaultState: MetaMetricsDataDeletionState = {
+  metaMetricsDataDeletionId: null,
+  metaMetricsDataDeletionTimestamp: 0,
+};
```

Comment:
nit: in some controllers we started preferring functions over variables for default states, to avoid changing the value at runtime:

```typescript
const getDefaultMetaMetricsDataDeletionState =
  (): MetaMetricsDataDeletionState => ({
    metaMetricsDataDeletionId: null,
    metaMetricsDataDeletionTimestamp: 0,
  });
```